### DISCUSSION TITLE
Help getting things right

### DISCUSSION THREAD

**Discussion by d3mac123** [schreibfaul1/ESP32-audioI2S#595] [General]
Hi,

I have a [Lilygo T-Display](https://github.com/Xinyuan-LilyGO/T-Display-S3) (ESP32 S3 model with no DAC pins). I need to play audio externally (all audio files are converted to .h files). Based on several searches, it seems this [product](https://www.amazon.com/AITRIP-Amplifier-Filterless-Amplifiers-Raspberry/dp/B0B4J93M9N/ref=asc_df_B0B4J93M9N/?tag=hyprod-20&linkCode=df0&hvadid=632101815305&hvpos=&hvnetw=g&hvrand=3495880550744405403&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=1015243&hvtargid=pla-1897361898641&th=1) may enable my board to output audio - if you have other suggestions, they are welcomed!

So far (while I don't have the DAC/amplifier module), I started writing some code but I don't know exactly how to add and play, for example, file "one.h" - anything I miss that someone can point me in the right direction?

Thank you all!

```
#include <Wire.h>
#include "Audio.h"
#include "one.h"
// MAX98357A I2S pins. 
#define I2S_WS 10
#define I2S_BCK 11
#define I2S_DOUT 12

Audio audio;

//** Here I need to connect with file one.h - Is the #include "one.h" line above enough? **

void setup() {
  Serial.begin(115200);
  Wire.begin(); 

  // Select I2S pins
  audio.setPinout(I2S_BCK, I2S_WS, I2S_DOUT);

  // Volume control.
  audio.setVolume(21);
}

void loop() {
  //** Here I need to play file one.h **
  audio.loop();
}
```

**Comment by schreibfaul1** [comment]
Yes, the S3 has no built-in DAC. Whereas the DAC of the ESP32 with its 8 bits (256 different analogue values) does not sound really good. The I2S output only delivers a digital signal.

**Comment by d3mac123** [comment]
I guess my questions are:
- can I use the solution I mentioned (quality of the audio is not super important - basically it is just a number counter - I could even using a TTS library, if it works easier)? https://www.amazon.com/AITRIP-Amplifier-Filterless-Amplifiers-Raspberry/dp/B0B4J93M9N/ref=asc_df_B0B4J93M9N/?tag=hyprod-20&linkCode=df0&hvadid=632101815305&hvpos=&hvnetw=g&hvrand=3495880550744405403&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=1015243&hvtargid=pla-1897361898641&th=1
- Is there any examples from your library that shows how to play an audio from a .h file?
